### PR TITLE
Show a "Targets" bar as part of the DocC generated header

### LINKED ISSUE #1743: Style DocC documentation links
The documentation links need styling up, both for single links and for packages with multiple documentation targets.

Screenshot 2022-05-18 at 15 51 06@2x

### PR DESCRIPTION
**Pull request by daveverwer**
Fixes #1743 

When there is > 1 documentation target:

Screenshot 2022-05-24 at 20 37 18@2x
Screenshot 2022-05-24 at 20 37 44@2x

When there is exactly 1 documentation target:

Screenshot 2022-05-24 at 20 38 23@2x
Screenshot 2022-05-24 at 20 38 32@2x

### CODE REVIEW

**Comment by daveverwer** [comment]
Looks like the bar isn't rendering correctly with Percy. Could you check it out locally and make sure it works, please, @finestructure?

**Comment by finestructure** [comment]
Looks great:

CleanShot 2022-05-25 at 09 24 09@2x

File: Sources/App/Views/DocumentationPageProcessor.swift [hunk 1]
```diff
@@ -22,16 +22,22 @@ struct DocumentationPageProcessor {
...
     let packageName: String
+    let reference: String
+    let targets: [String]
 
     init?(repositoryOwner: String,
           repositoryOwnerName: String,
           repositoryName: String,
           packageName: String,
+          targets: [String],
+          reference: String,
```
Comment (finestructure) [inline Sources/App/Views/DocumentationPageProcessor.swift:33]: nit: reverse order to other two additions
```suggestion
          reference: String,
          targets: [String],
```

File: Sources/App/Views/PackageController/PackageShow+Model.swift [hunk 2]
```diff
@@ -176,6 +176,7 @@ extension PackageShow.Model {
         "https://github.com/\(repositoryOwner)/\(repositoryName)"
     }
 
+    // Note: When this gets merged back with the refactored SiteURL, note that it's duplicated in `DocumentationPageProcessor`.
```
Comment (finestructure) [inline Sources/App/Views/PackageController/PackageShow+Model.swift:179]: Probably better than to just make it a static func on `SiteURL` for now. That way we'll address it automatically.

I'll take a look.

Comment (finestructure) [reply Sources/App/Views/PackageController/PackageShow+Model.swift:179]: Never mind, didn't see the additional parameters `repositoryOwner` and `repositoryName` being used.

**Comment by daveverwer** [comment]
> Looks great:

Thanks. I'll log this as a separate issue (#1761) so we don't hold this PR up as I won't have time to look at this until next week.

**Review by finestructure — APPROVED** [review]

**Pull request merged**